pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -898,14 +899,35 @@ func possibleGuestSize(disk api.Disk) (int64, bool) {
 		return 0, false
 	}
 
-	size := int64((1 - filesystemOverhead) * float64(preferredSize))
+	usableSize, err := getUsableDiskSize(getSourceFile(disk))
```

Comment:
So this won't work for certain provisioners. For instance the hostpath provisioner. Since you will use up all the space available. You should take min(preferredSize, usableSize) and use that instead.